scripts/check-pr.sh

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env bash
+
+# This script is executed by Travis CI for every pull request opened.
+# It currently accomplishes the following objectives (for English pages only):
+#
+#  1. Detect pages that were just copied (i.e. cp pages/{common,linux}/7z.md).
+#  2. Detect pages that were added in a platform specific directory although
+#     they already exist under 'common'.
+#  3. Detect pages that were added in the 'common' platform although they
+#     already exist under a platform specific directory.
+#  4. Detect other miscellaneous anomalies in the pages folder.
+#
+# Results are printed to stdout, logs and errors to stderr.
+#
+# NOTE: must be run from the repository root directory to correctly work!
+# NOTE: no `set -e`, failure of this script should not invalidate the build.
+
+# Check for duplicated pages.
+function check_duplicates {
+  local msg='The page `%s` already exists under the `%s` platform.\n'
+  local page=$1 # page path in the format 'platform/pagename.md'
+  local parts
+  local other
+
+  readarray -td'/' parts < <(echo -n "$page")
+
+  local platform=${parts[0]}
+  local file=${parts[1]}
+
+  case "$platform" in
+    common) # check if page already exists in other platforms
+      for other in `ls pages/`; do
```

Comment:
Let's compute `ls pages/` from before and reuse it. This is being executed repeteadly.